### ISSUE TITLE
custom downstream RKE won't fully register from behind http proxy

### ISSUE THREAD

**Issue by jmozd** [rancher/rancher#35939]
**Rancher Server Setup**
- Rancher version: 2.6.3 (original report for Rancher 2.6.2, but upgraded Rancher since)
- Installation option (Docker install/Helm Chart): Helm Chart, RKE1

**Information about the Cluster**
- Kubernetes version: v1.21.7 on docker://20.10.9-ce on SUSE Linux Enterprise Server 15 SP2
- Cluster Type (Local/Downstream): Downstream RKE1 via RMS (custom install)

**Describe the bug**
I'm having trouble bringing up a custom RKE cluster deployed via Rancher. This seems to be related to the downstream needing an HTTPS_PROXY to reach the Rancher Management Server, happens since upgrading RMS to 2.6.2 and affects "share-mnt" containers on all nodes:

- configuring five nodes with SLES 15 SP2 (JeOS image in Openstack Cluster)
- configuring custom RKE1 cluster in RMS, adding HTTP_PROXY, HTTPS_PROXY, NO_PROXY env vars
- running the registration command on the five nodes (3x etc and controlplane, 2x worker). The RMS-generated registration command does already list the HTTP_PROXY, HTTPS_PROXY and NO_PROXY env vars, so the configuration of the downstream cluster in RMS should be ok.
- stage one of node setup runs fine AFAICT - the cluster gets up at Kubernetes level, kubectl works nicely
- RMS shows the cluster and 5 machines as "active", but "Explore" button remains gray
- "docker ps" on all five nodes shows the share_mnt container getting started again and again, with the log entries
```
INFO: Environment: CATTLE_ADDRESS=192.168.163.39 CATTLE_AGENT_CONNECT=true CATTLE_INTERNAL_ADDRESS= CATTLE_NODE_NAME=kube05 CATTLE_SERVER=https://rancher.cloud.customer.com CATTLE_TOKEN=REDACTED CATTLE_WRITE_CERT_ONLY=true
INFO: Using resolv.conf: search customer.com cloud.customer.com nameserver 192.168.163.17 nameserver 192.168.163.11 nameserver 192.168.163.10
ERROR: https://rancher.cloud.customer.com/ping is not accessible (Failed to connect to rancher.cloud.customer.com port 443: Connection timed out)
```
tcpdump on the nodes show traffic to RMS 443 port directly (which should not happen, but rather go via external Squid proxy set in HTTPS_PROXY).

In other words, it looks like the share_mnt container is started by rancher-agent without passing the required env vars for proxy access.

**To Reproduce**
Install downstream cluster with HTTP_PROXY etc env vars. Reproducible for me 100%

**Result**
share_mnt container continuesly timing out and restarting

**Expected Result**
node registration should complete

**Comment by jmozd** [comment]
reproducible with Rancher 2.6.3 (upgraded via Helm from 2.6.2)

**Comment by FrediWeber** [comment]
We can confirm this issue - also happening in our environment.

**Comment by jmozd** [comment]
After a complete reinstall of the Rancher server v2.6.3-patch1 (not only the downstream clusters), I cannot reproduce the behavior.
Until then, I had been testing with my Rancher server that was upgraded from 2.5.7 onward, and which has suffered a number of "cannot delete downstream cluster" incidents (fixed by manually deleting "finalizers"). Maybe something got screwed up in Ranchers data base.

@FrediWeber: Do you mind if I close this issue, since I can no longer reproduce?

**Comment by jmozd** [comment]
please ignore my last comment.

I had a large number of problems with latest installs, which were gone once I reinstalled Rancher as mentioned above. I now have a situation where after the installation of downstream clusters (which need an HTTP proxy to reach the Rancher server), they are manageable from Rancher (shown as "active", "explore" button in web UI is active and functional).

Nevertheless, on the worker nodes, I see that the share-mnt container is constantly restarting, with the original error in the logs:

ERROR: https://rancher.cloud.customer.com/ping is not accessible (Failed to connect to rancher.cloud.hh.nde.ag port 443: Connection timed out)

Since the Rancher server "rancher.cloud.customer.com" is only accessible via HTTP proxy, it seems obvious that the share-mnt container still is missing the HTTP_PROXY and NO_PROXY variables, which should have been inherited from the initial Rancher container.

Once I reconfigure networking so that the node can reach the Rancher server without proxy, share_mnt completes.
